borg/archiver.py

```diff
@@ -371,12 +381,11 @@ def do_extract(self, args, repository, manifest, key, archive):
         sparse = args.sparse
         strip_components = args.strip_components
         dirs = []
-        for item in archive.iter_items(lambda item: matcher.match(item[b'path']), preload=True):
+        filter = self.build_filter(matcher, strip_components)
+        for item in archive.iter_items(filter, preload=True):
             orig_path = item[b'path']
             if strip_components:
                 item[b'path'] = os.sep.join(orig_path.split(os.sep)[strip_components:])
-                if not item[b'path']:
-                    continue
```

Comment:
So, the problem this change avoids has its roots in archive.iter_items (or code called from there)?